cpp/autosar/test/rules/M9-3-3/test.cpp

```diff
@@ -161,3 +161,29 @@ class Z22 : Z1 {
   void f2() final {}                   // COMPLIANT
   void f3() { this->a = 100; }         // COMPLIANT
 };
+
+template <class T> class Array {
+public:
+  T &back();
+
+private:
+  T data[128];
+  unsigned int size;
+};
+
+template <class T, template <class...> class U> class Stack {
+public:
+  T &Top() {
+    return this->data.back();
+  } // Likely NON_COMPLIANT, but cannot be determined until instantiation.
```

Comment:
lets change this message to best describe the expected behaviour (align with the [style guide](https://github.com/github/codeql-coding-standards/blob/main/docs/development_handbook.md#style-guide)), 

in this case with the instantiation, it is a COMPLIANT example due to (after instantiation) being able to see that it calls a non const using member variable.

though this location represents 2 member functions, and one (the instantiated one) is compliant due to the logic of the rule
and the non instantiated one is compliant due to not having enough info to know if it obeys the rule - 

so maybe... label COMPLIANT - instantiated compliant truly and uninstantiated compliant via exception
or whatever seems most clear to describe the expected behaviour

Reply:
I would argue that the style guide is inconclusive about cases that are neither compliant nor non-compliant. Perhaps we should call it `COMPLIANT[FALSE_NEGATIVE|TRUE_NEGATIVE]` to highlight a nonstandard exception with the explanation that the template dictates the interface and not its instantiations?

Reply:
I would be happy with either a `COMPLIANT - exclude consideration of functions in uninstantiated templates` or `COMPLIANT[FALSE_NEGATIVE|TRUE_NEGATIVE] - insert explanation` with the explanation you mention

its true that the latter doesnt completely follow the style guide, so I am leaning towards the former but would find either to communicate what is going on well enough (at least imo labelling it both fn/tn indicates that we are purposely ignoring it)